mlrun/api/crud/pipelines.py

```diff
@@ -102,6 +98,38 @@ def list_pipelines(
 
         return total_size, next_page_token, runs
 
+    def delete_pipelines_runs(self, db_session: sqlalchemy.orm.Session, project: str):
+        _, _, project_pipeline_runs = self.list_pipelines(
+            db_session=db_session,
+            project=project,
+            format_=mlrun.common.schemas.PipelinesFormat.metadata_only,
+        )
+        kfp_client = self.initialize_kfp_client()
+
+        if project_pipeline_runs:
+            logger.debug(
+                "Detected pipeline runs for project, deleting them",
+                project=project,
+                pipeline_runs=[run["id"] for run in project_pipeline_runs],
+            )
+
+        for pipeline_run in project_pipeline_runs:
+            try:
+                # delete pipeline run also terminates it if it is in progress
+                kfp_client._run_api.delete_run(pipeline_run["id"])
+            except Exception as exc:
+                # we don't want to fail the entire delete operation if we failed to delete a single pipeline run
+                # (we will log the error and continue) because it is being used as part of the delete project
+                # operation and we don't want to fail the entire delete project operation because of a single
+                # pipeline run
```

Comment:
No need to write it twice
```suggestion
                # we don't want to fail the entire delete operation if we failed to delete a single pipeline run
                # so it won't fail the delete project operation. we will log the error and continue
```